developmentseed/osm-seed
# Ingress ssl

## What I am changing

See #216 - adding auto-configuration of domains and SSL for services on the cluster.


## How you can test it

This is a bit involved to test. 

Currently in the PR, I have only set this up for the `web` service, but we would want to replicate this for other services once we are happy with the setup for `web`.

To test this, you would need to:

 - Setup a new cluster
 - Follow the instructions added in this PR to set the nginx ingress controller and `cert-manager` via helm.
 - Set a value for `domain` in values.yaml . Say we set `osmseed-test.ds.io` .
 - Get the ClusterIP for the cluster by doing `kubectl get ingress` or so should give the Cluster IP
 - Create an A record for `*.osmseed-test.ds.io` to point to the cluster IP
 - Test that https://web.osmseed-test.ds.io works in the browser and that SSL is setup correctly.
 - If it does not work, one will need to check kubernetes logs to try and figure out what is failing

Once we test and manage to make this work, we should decide on a good pattern to implement this across all services.

Some considerations:

 - We should make sure it still works to deploy to `minikube` by not requiring this for a minikube install
 - Consider if we want to still allow users to specify type=LoadBalancer and handle routing and SSL themselves, and then it won't try and do any of this Ingress and SSL stuff.

@Rub21 @geohacker - let's chat about this on scrum today and see if there's a good way to try and test this, and then discuss what pattern we are happy with, and then implement the same for the rest of the services.

## Related Issues

#216
